src/events/broker.py

```diff
@@ -423,22 +417,51 @@ def healthy(self) -> bool:
 
         return True
 
-    def _create_internal_credentials(self) -> list[tuple[str, str]]:
-        """Creates internal SCRAM users during cluster start.
-
-        Returns:
-            List of (username, password) for all internal users
+    def _init_kraft_mode(self) -> None:
+        """Initialize the server when running controller mode."""
+        # NOTE: checks for `runs_broker` in this method should be `is_cluster_manager` in
+        # the large deployment feature.
+        if self.model.unit.is_leader():
+            if (
+                not self.charm.state.cluster.internal_user_credentials
+                and self.charm.state.runs_broker
+            ):
+                credentials = [
+                    (username, self.charm.workload.generate_password())
+                    for username in INTERNAL_USERS
+                ]
+                for username, password in credentials:
+                    self.charm.state.cluster.update({f"{username}-password": password})
```

Comment:
**todo:** Should this be part of `_init_kraft_mode` and not separate? Or is it bundled here because we're branching Zk-mode and kraft-mode, so no need to support `create_internal_credentials` for ZK anymore?